manager/orchestrator/restart/restart.go

```diff
@@ -190,6 +192,83 @@ func (r *Supervisor) Restart(ctx context.Context, tx store.Tx, cluster *api.Clus
 	return nil
 }
 
+// TaskRestartDelay calculates and returns the next delay based t's RestartPolicy
+func (r *Supervisor) TaskRestartDelay(ctx context.Context, t *api.Task) (*time.Duration, error) {
```

Comment:
Still need to change this so it doesn't return a pointer. You can just `return 0, errors.New("whatever")` if there is an error. Go convention is to not look at/use the first return value if there is an error, so it doesn't really matter what the return value is, but the 0-value for the type is conventional (and the 0-value of `time.Duration` is 0).